### DISCUSSION TITLE
Caused by: Failed to build a native library through cargo

### DISCUSSION THREAD

**Discussion by reallyusefulengine** [PyO3/maturin#1018] [Q&A] [answered]
Im following the book refactoring to rust which uses maturin. 

When i create a virtual env and activate and run the following command I get an error:

```
py_rust % maturin develop
🔗 Found pyo3 bindings
🐍 Found CPython 3.9 at /Users/johnwilliams/.local/share/virtualenvs/py_rust-QUdKXKyu/bin/python
💥 maturin failed
  Caused by: Failed to build a native library through cargo

```

im using an m1 mac and my folder structure looks like this:
<img width="1619" alt="image" src="https://user-images.githubusercontent.com/45160568/179990661-cf81d77a-bd5b-4e96-8c72-66487bde8222.png">

Any help would be appreciated to get me blocked.

**Comment by messense** [comment, marked as answer]
Please fill out the issue form.

https://github.com/PyO3/maturin/issues/1019#issuecomment-1190281767
